### DISCUSSION TITLE
CSS seems not to load if using a kemp load balancer

### DISCUSSION THREAD

**Discussion by kacoroski** [YOURLS/YOURLS#3195] [Q&A]
Hi,

Wondering if anyone may know what is happening.  The yourls site works fine when I go directly to it, but when I put a kemp load balancer in front of it to do ssl offloading, it seems like it is not loading the CSS as I do not get a background color or the nice boxes and layout of the site.  Any ideas as to what could be happening.  Thanks in advance.

cheers,

ski

**Comment by dgw** [comment]
Server and/or browser DevTools logs are going to be able tell you _way_ more (read: anything) than one of us can guess. TLS/mixed-content issues, for example, will almost always show up in the browser console log.

**Comment by kacoroski** [comment]
Urgh, I should have done that.  Mixed content is the problem:

Mixed Content: The page at 'https://kgo.nsd.org/' was loaded over HTTPS, 
but requested an insecure stylesheet 
'http://yourls.nsd.org/css/style.css?v=1.8.2'. This request has been 
blocked; the content must be served over HTTPS.
kgo.nsd.org/:12 Mixed Content: The page at 'https://kgo.nsd.org/' was 
loaded over HTTPS, but requested an insecure stylesheet 
'http://yourls.nsd.org/css/tablesorter.css?v=1.8.2'. This request has 
been blocked; the content must be served over HTTPS.
kgo.nsd.org/:16 Mixed Content: The page at 'https://kgo.nsd.org/' was 
loaded over HTTPS, but requested an insecure stylesheet 
'http://yourls.nsd.org/css/share.css?v=1.8.2'. This request has been 
blocked; the content must be served over HTTPS.
kgo.nsd.org/:19 Mixed Content: The page at 'https://kgo.nsd.org/' was 
loaded over HTTPS, but requested an insecure stylesheet 
'http://yourls.nsd.org/css/cal.css?v=1.8.2'. This request has been 
blocked; the content must be served over HTTPS.

Is the best way to fix this for me to edit the yourls php files and 
replace http with https everywhere or is there a single setting that I 
can set to fix this.

cheers,

ski

On 1/13/22 2:47 PM, dgw wrote:
> Server and/or browser DevTools logs are going to be able tell you /way/ 
> more (read: anything) than one of us can guess. TLS/mixed-content 
> issues, for example, will almost always show up in the browser console log.
> 
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/YOURLS/YOURLS/discussions/3195#discussioncomment-1966027>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAMWMAKYESQOKPSG5G4FUWTUV5I73ANCNFSM5L5EOZAA>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
> 
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>
> 

-- 
"When we try to pick out anything by itself, we find it
   connected to the entire universe"            John Muir

Chris "Ski" Kacoroski, ***@***.***, 206-501-9803

**Comment by kacoroski** [comment]
Ok, got it fixed.  The kemp has a neat trick that where I can replace 
all http with https in the responses.  Works great.

cheers,

ski

On 1/13/22 2:47 PM, dgw wrote:
> Server and/or browser DevTools logs are going to be able tell you /way/ 
> more (read: anything) than one of us can guess. TLS/mixed-content 
> issues, for example, will almost always show up in the browser console log.
> 
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/YOURLS/YOURLS/discussions/3195#discussioncomment-1966027>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAMWMAKYESQOKPSG5G4FUWTUV5I73ANCNFSM5L5EOZAA>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
> 
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>
> 

-- 
"When we try to pick out anything by itself, we find it
   connected to the entire universe"            John Muir

Chris "Ski" Kacoroski, ***@***.***, 206-501-9803

**Reply by dgw** [reply]
That's one way, or if you enforce HTTPS at the load balancer layer you can set `YOURLS_SITE` to include the `https://` and that should also work.
